daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/ProcessorBases.scala

```diff
@@ -68,10 +68,24 @@ trait Processor
   def runtimeDependencies: Seq[Evaluatable[AnyRef]]
 
   var isInitialized: Boolean = false
+
+  def isPrimitive: Boolean
 }
 
 trait PrimProcessor extends Processor {
   override def childProcessors: Seq[Processor] = Nil
+
+  override def isPrimitive = true
```

Comment:
Scaladoc for these traits, and methods.

The whole point of forcing every parser/unparser to be classified as one of Prim, PrimNoData, or Combinator is because things like bitOrder only need to be considered for prim parsers. If we can recognize a prim parser, then we can centralize checking of things like the bitOrder in the PState/UState implementation of FormatInfo.